public/framework/catalog_cache/cache_loader.tsx

```diff
@@ -123,11 +129,53 @@ export const updateAccelerationsToCache = (
   });
 };
 
+export const updateTableColumnsToCache = (
+  dataSourceName: string,
+  databaseName: string,
+  tableName: string,
+  pollingResult: AsyncPollingResult
+) => {
+  const cachedDatabase = CatalogCacheManager.getDatabase(dataSourceName, databaseName);
+  const currentTime = new Date().toUTCString();
+
+  if (!pollingResult) {
```

Comment:
Why not move this to the first line of the method? I would assume `CatalogCacheManager.getDatabase` is expensive, and `pollingResult` doesn't rely on that, so wouldn't it be better to quickly exit if there's no result yet?

Reply:
good catch! updated this.